pennylane/_qubit_device.py

```diff
@@ -178,12 +175,17 @@ def rotate_basis(self, obs_queue):
                 else:
                     wires.extend(wire)
 
+        # Store the wires used by the observables such that
+        # an Identity is considered on the remaining wires
         self._wires_used = wires
         self._rotated_prob = self.probability(wires)
 
     def generate_samples(self):
         """Generate computational basis samples based on the current state.
 
+        Warning: this method will have to be redefined for hardware devices, since it uses
+        the device._rotated_prob. This attribute might not be available for such devices.
```

Comment:
Note: you can use `.. warning::` here to make a nice yellow box. Also, I would rephrase this - `device._rotated_prob` is set internally by the `QubitDevice` by simply calling `self.probability`. The reason this should be redefined for hardware devices: because hardware devices generate samples themselves :)

Reply:
Ah you are right, thanks! Also, I just saw that there is already a warning included down below, so I'll just leave that as included (instead of having a new one defined :smile:).